### DISCUSSION TITLE
REST API SDKs!

### DISCUSSION THREAD

**Discussion by Ymirke** [twentyhq/twenty#6281] [Engineering vision]
Building integrations towards twenty would be much faster if there was a typescript REST api client.

I really want types and auto complete for api calls. Would be magical.

**Comment by deezeddd** [comment]
That's a great idea! are you working on this currently?

**Comment by FelixMalfait** [comment]
Should be fairly easy to do with the OpenAPI spec and a tool like this https://heyapi.vercel.app/openapi-ts/get-started.html

One reason we don't provide it is that the api is different for each workspace.

But by end of year we'll add the ability to code within Twenty and probably we will leverage the above tool to generate one SDK per workspace and make it available in the development environment's context

**Reply by deezeddd** [reply]
Yup, the plan looks great @FelixMalfait. I would love to work on building the SDK. Kindly let me know when you plan to work upon it. I’ll be more than happy to help :)

**Comment by BOHEUS** [comment]
Core team is currently working on extensibility and most likely this quarter SDK will be stable to the point of creating actual apps (SDK uses GraphQL calls under the hood)
